controllers/managedocs_controller_test.go

```diff
@@ -852,7 +861,41 @@ var _ = Describe("ManagedOCS controller", func() {
 				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
 			})
 		})
-		When("there is a value for PAGERDUTY_KEY in the pagerduty secret and a value for SNITCH_URL in the deadmanssnitch secret", func() {
+		When("there is a notification email address in the add-on parameter secret and no smtp secret", func() {
+			It("should not create alertmanager config", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secretKey := utils.GetResourceKey(secret)
+				Expect(k8sClient.Get(ctx, secretKey, secret)).Should(Succeed())
+
+				// Add notification email to the addon param secret
+				secret.Data["notification-email-0"] = []byte("test@email.com")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				// Create pagerduty secret with valid key
+				pdSecret := pdSecretTemplate.DeepCopy()
+				pdSecret.Data["PAGERDUTY_KEY"] = []byte("test-key")
+
+				// Create deadman's snitch secret with a valid key
+				dmsSecret := dmsSecretTemplate.DeepCopy()
+				dmsSecret.Data["SNITCH_URL"] = []byte("test-key")
+
+				Expect(k8sClient.Create(ctx, pdSecret)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, dmsSecret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resources are not created
+				utils.EnsureNoResource(k8sClient, ctx, amConfigTemplate.DeepCopy(), timeout, interval)
+
+				// Remove the secret for future cases
+				Expect(k8sClient.Delete(ctx, pdSecret)).Should(Succeed())
+				Expect(k8sClient.Delete(ctx, dmsSecret)).Should(Succeed())
+
+				// Remove notification email for future cases
+				delete(secret.Data, "notification-email-0")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+			})
+		})
+		When("there is a value for PAGERDUTY_KEY in the pagerduty secret, a value for SNITCH_URL in the deadmanssnitch secret and a SMTP secret", func() {
```

Comment:
Each test should isolate one check, this one seems to me like an aggregation of many checks. 
If this is check is supposed to test a situation where all alert manager conditions are met, let's describe it as such. 

Think. about the case where we will introduce many more conditions, we will not want to track all of them in the description for this case. 

```suggestion
		When("All conditions for creating an alertmanager config are met ", func() {
```